File: hedera-sdk-go/examples/create-account.go
package main

import (
	"fmt"
	"github.com/mehcode/hedera-sdk/hedera-sdk-go"
	"os"
	"time"
)

func main() {
	//
	// Generate keys
	//

	// Read and decode the operator secret key
	operatorSecret := hedera.SecretKeyFromString(os.Getenv("OPERATOR_SECRET"))

	// Generate a new keypair for the new account
	secret := hedera.GenerateSecretKey()
	public := secret.Public()

	fmt.Printf("secret = %v\n", secret)
	fmt.Printf("public = %v\n", public)

	//
	// Connect to Hedera
	//

	client := hedera.Dial("testnet.hedera.com:50001")
	// TODO: client.SetRetryOnFailure(0) // default: 5
	defer client.Close()

	//
	// Send transaction to create account
	//

	nodeAccountId := hedera.NewAccountID(0, 0, 3)
	operatorAccountID := hedera.NewAccountID(0, 0, 2)
	response := client.CreateAccount().
		Operator(operatorAccountID).
		Node(nodeAccountId).
		Key(public).
		InitialBalance(0).
		Memo("[test] hedera-sdk-go v2").
		Sign(operatorSecret).
		Execute()

	transactionID := response.ID
	fmt.Printf("created account; transaction = %v\n", transactionID)

	//
	// Get receipt to prove we created it ok
	//

	fmt.Printf("wait for 2s...\n")
	time.Sleep(2 * time.Second)

	receiptResponse := client.GetTransactionReceipt(transactionID).Send()

	if receiptResponse.Precheck != 0 {
		fmt.Printf("receiptResponse:pre-check != OK (%v)\n", receiptResponse.Precheck)
		return
	}

	if receiptResponse.Receipt.AccountID != nil {
		// TODO: Add pretty printing for account ID
		fmt.Printf("account = %v\n", *receiptResponse.Receipt.AccountID)
	}
}

File: examples/create_account.rs
use failure::{format_err, Error};
use futures::FutureExt;
use hedera::{Client, SecretKey, Status};
use std::{env, thread::sleep, time::Duration};

#[tokio::main]
async fn main() -> Result<(), Error> {
    pretty_env_logger::try_init()?;

    let (secret, _) = SecretKey::generate("");
    let public = secret.public();

    println!("secret = {}", secret);
    println!("public = {}", public);

    // Operator is the account that sends the transaction to the network
    // This account is charged for the transaction fee
    let operator = "0:0:2".parse()?;
    let client = Client::builder("testnet.hedera.com:50131")
        .node("0:0:3".parse()?)
        .operator(operator, || env::var("OPERATOR_SECRET"))
        .build()?;

    // Create our account
    let id = client
        .create_account()
        .key(public)
        .initial_balance(5_000_000)
        .memo("[hedera-sdk-rust][example] create_account")
        .execute_async()
        .await?;

    println!("created account; transaction = {}", id);

    // If we got here we know we passed pre-check
    // Depending on your requirements that may be enough for some kinds of transactions
    sleep(Duration::from_secs(2));

    // Get the receipt and check the status to prove it was successful
    let mut tx = client.transaction(id).receipt();
    let receipt = tx.get_async().await?;

    if receipt.status != Status::Success {
        Err(format_err!(
            "transaction has a non-successful status: {:?}",
            receipt.status
        ))?;
    }

    // note: account can be [None] if the receipt wasn't for creating an account
    let account = receipt.account_id.unwrap();
    println!("account = {}", account);

    Ok(())
}
